### PR TITLE
Revised the enumeration types article

### PR DESCRIPTION
**Pull request by pkulikov**
Follow up of the comment in https://github.com/dotnet/docs/pull/16166#pullrequestreview-329858080
Contributes to #10726

### CODE REVIEW

File: docs/csharp/language-reference/builtin-types/enum.md [hunk 1]
```diff
@@ -38,40 +38,35 @@ enum ErrorCode : ushort
...
->
-> If you explicitly specify associated values of enum members and want to ensure that the default value of an enumeration type is a declared value, declare an enum member with the associated value 0.
+The default value of an enumeration type `E` is the value produced by expression `(E)0`, even if zero doesn't have the corresponding enum member.
 
-The <xref:System.Enum?displayProperty=nameWithType> type is the abstract base class of all enumeration types. It provides a number of methods to get information about an enumeration type and its values. For more information and examples, see the <xref:System.Enum?displayProperty=nameWithType> API reference page.
-
-Beginning with C# 7.3, you can use `System.Enum` in a base class constraint (that is known as the [enum constraint](../../programming-guide/generics/constraints-on-type-parameters.md#enum-constraints)) to specify that a type parameter is an enumeration type.
+You use an enumeration type to represent a choice from a set of mutually exclusive values or a combination of choices. To be able to represent a combination of choices, define an enumeration type as bit flags.
```
Comment (BillWagner) [inline docs/csharp/language-reference/builtin-types/enum.md]: nit:

```suggestion
You use an enumeration type to represent a choice from a set of mutually exclusive values or a combination of choices. To represent a combination of choices, define an enumeration type as bit flags.
```

File: docs/csharp/language-reference/builtin-types/enum.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ ms.assetid: bbeb9a0f-e9b3-41ab-b0a6-c41b1a08974c
 ---
 # Enumeration types (C# reference)
 
-An enumeration type (or enum type) is a value type that represents a set of named integral constants. To define an enumeration type, use the `enum` keyword and specify the names of *enum members*:
+An enumeration type (or enum type) is a value type that is defined by a set of named constants of the underlying [integral numeric](integral-numeric-types.md) type. To define an enumeration type, use the `enum` keyword and specify the names of *enum members*:
```
Comment (BillWagner) [inline docs/csharp/language-reference/builtin-types/enum.md]: nit:

```suggestion
An enumeration type (or enum type) is a value type defined by a set of named constants of the underlying [integral numeric](integral-numeric-types.md) type. To define an enumeration type, use the `enum` keyword and specify the names of *enum members*:
```

File: docs/csharp/language-reference/builtin-types/enum.md [hunk 3]
```diff
@@ -38,40 +38,35 @@ enum ErrorCode : ushort
...
 
-## Enumeration types as bit flags
+## Conversions
 
-You typically use an enumeration type to represent a set of mutually exclusive values. However, in some scenarios, a combination of enumeration values can also be a meaningful enumeration value. Then, you can define only those enum members that represent a bit field in an enumeration value and use the [bitwise logical operators](../operators/bitwise-and-shift-operators.md#enumeration-logical-operators) to combine enumeration values. The associated values of the enum members that represent a bit field should be the powers of two. To indicate that an enumeration type declares bit fields, apply the [Flags](xref:System.FlagsAttribute) attribute to it. The following example defines the `Days` enumeration type and shows the behavior of the bitwise operators with its instances.
+For an enumeration type `E` with an underlying type `U`, there exist explicit conversions from `E` to `U` and from `U` to `E`. If you [cast](../operators/type-testing-and-cast.md#cast-operator-) an enum value of type `E` to an underlying type `U`, the result is the associated integral value of an enum member.
```
Comment (BillWagner) [inline docs/csharp/language-reference/builtin-types/enum.md]: I'd say this a little less like the spec language:

```suggestion
For any enumeration type, there exist explicit conversions between the enumeration type and its underlying intergral type. If you [cast](../operators/type-testing-and-cast.md#cast-operator-) an enum value to its underlying type, the result is the associated integral value of an enum member.
```

**Review by BillWagner — CHANGES_REQUESTED** [review]
This is great work @pkulikov 

I had a couple suggestions for wording to make this article a bit less like the spec language. 

Then, it's ready to merge.

**Comment by pkulikov** [comment]
@BillWagner thanks! I've applied all of your suggestions.

**Review by BillWagner — APPROVED** [review]

**Pull request merged**